lumicks/pylake/detail/confocal.py

```diff
@@ -6,22 +6,38 @@
 from .mixin import PhotonCounts
 from .mixin import ExcitationLaserPower
 from .image import reconstruct_image_sum, reconstruct_image, save_tiff, ImageMetadata
+from .utilities import could_sum_overflow
+
+
+def _int_mean(a, total_size, axis):
+    if not could_sum_overflow(a, axis):
+        return np.sum(a, axis) // total_size
+
+    n = a.size // 2 if axis is None else a.shape[axis] // 2
+    b = a if axis is None else np.swapaxes(a, axis, 0)
+    return _int_mean(b[:n], total_size, 0) + _int_mean(b[n:], total_size, 0)
```

Comment:
You could flip these:
```suggestion
    b = a if axis is None else np.swapaxes(a, axis, 0)
    n = b.shape[0] // 2
    return _int_mean(b[:n], total_size, 0) + _int_mean(b[n:], total_size, 0)
```

Reply:
ah that's a nice way to get rid of the ternary :)